### PR TITLE
add gradual ponzi

### PR DESCRIPTION
**Pull request by konnov**
One more example that may be useful for the workshops.

- [x] Tests added for any new code

### CODE REVIEW

File: examples/solidity/GradualPonzi/gradualPonzi.qnt [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    if (amount > evm.balances.get(investor)) {
+      error: "Insufficient funds",
+      evm: evm,
+      ponzi: ponzi,
+     } else if (amount < MINIMUM_INVESTMENT) {
+      error: "Investment below minimum",
+      evm: evm,
+      ponzi: ponzi,
+    } else
+      pure val investorCut = amount / length(ponzi.investors)
```
Comment (bugarela) [inline examples/solidity/GradualPonzi/gradualPonzi.qnt:62]: Can this integer division be changing the total supply?

**Review by bugarela — APPROVED** [review]
LGTM, just a few comments

File: examples/solidity/GradualPonzi/gradualPonzi.qnt (hunk 1 above)
Comment (konnov) [reply examples/solidity/GradualPonzi/gradualPonzi.qnt:65]: I have added `noLeftoversInv`, which shows that the contract may accumulate leftover tokens due to rounding. It's actually quite nice.

Comment (bugarela) [reply examples/solidity/GradualPonzi/gradualPonzi.qnt:65]: Awesome, makes sense to me!

**Pull request merged**